validate/validate.go

```diff
@@ -0,0 +1,574 @@
+	for os, archs := range validCombins {
+		if os == platform.OS {
+			for _, arch := range archs {
+				if arch == platform.Arch {
+					return nil
+				}
+			}
+			msgs = append(msgs, fmt.Sprintf("Combination of %q and %q is invalid.", platform.OS, platform.Arch))
+		}
+	}
+	msgs = append(msgs, fmt.Sprintf("Operation system %q of the bundle is not supported yet.", platform.OS))
+
+	return
+}
+
+func checkEventHooks(hookType string, hooks []rspec.Hook, hostSpecific bool) (msgs []string) {
+	for _, hook := range hooks {
+		if !filepath.IsAbs(hook.Path) {
+			msgs = append(msgs, fmt.Sprintf("The %s hook %v: is not absolute path", hookType, hook.Path))
+		}
+
+		if hostSpecific {
+			fi, err := os.Stat(hook.Path)
+			if err != nil {
+				msgs = append(msgs, fmt.Sprintf("Cannot find %s hook: %v", hookType, hook.Path))
+			}
+			if fi.Mode()&0111 == 0 {
+				msgs = append(msgs, fmt.Sprintf("The %s hook %v: is not executable", hookType, hook.Path))
+			}
+		}
+
+		for _, env := range hook.Env {
+			if !envValid(env) {
+				msgs = append(msgs, fmt.Sprintf("Env %q for hook %v is in the invalid form.", env, hook.Path))
+			}
+		}
+	}
+
+	return
+}
+func (v *Validator) CheckHooks() (msgs []string) {
+	logrus.Debugf("check hooks")
+
+	msgs = append(msgs, checkEventHooks("pre-start", v.spec.Hooks.Prestart, v.hostSpecific)...)
+	msgs = append(msgs, checkEventHooks("post-start", v.spec.Hooks.Poststart, v.hostSpecific)...)
+	msgs = append(msgs, checkEventHooks("post-stop", v.spec.Hooks.Poststop, v.hostSpecific)...)
+
+	return
+}
+
+func (v *Validator) CheckProcess() (msgs []string) {
+	logrus.Debugf("check process")
+
+	process := v.spec.Process
+	if !path.IsAbs(process.Cwd) {
+		msgs = append(msgs, fmt.Sprintf("cwd %q is not an absolute path", process.Cwd))
+	}
+
+	for _, env := range process.Env {
+		if !envValid(env) {
+			msgs = append(msgs, fmt.Sprintf("env %q should be in the form of 'key=value'. The left hand side must consist solely of letters, digits, and underscores '_'.", env))
+		}
+	}
+
+	for index := 0; index < len(process.Capabilities); index++ {
+		capability := process.Capabilities[index]
+		if !capValid(capability) {
+			msgs = append(msgs, fmt.Sprintf("capability %q is not valid, man capabilities(7)", process.Capabilities[index]))
+		}
+	}
+
+	for index := 0; index < len(process.Rlimits); index++ {
+		if !rlimitValid(process.Rlimits[index].Type) {
+			msgs = append(msgs, fmt.Sprintf("rlimit type %q is invalid.", process.Rlimits[index].Type))
+		}
+	}
+
+	if len(process.ApparmorProfile) > 0 {
+		profilePath := filepath.Join(v.rootfs, "/etc/apparmor.d", process.ApparmorProfile)
+		_, err := os.Stat(profilePath)
+		if err != nil {
+			msgs = append(msgs, err.Error())
+		}
+	}
+
+	return
+}
+
+func supportedMountTypes(OS string, hostSpecific bool) (map[string]bool, error) {
+	supportedTypes := make(map[string]bool)
+
+	if OS != "linux" && OS != "windows" {
+		logrus.Warnf("%v is not supported to (v *Validator) Check mount type", OS)
```

Comment:
Rouge find/replace?  The old warning was:

> "%v is not supported to check mount type"

I don't see a need to mess with that in this PR.